homeassistant/components/sensor/fido.py

```diff
@@ -142,21 +143,35 @@ def icon(self):
         """Icon to use in the frontend, if any."""
         return self._icon
 
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            'number': self._number,
+        }
+
     def update(self):
         """Get the latest data from Fido and update the state."""
         self.fido_data.update()
-        if self.type in self.fido_data.data:
-            if self.fido_data.data[self.type] is not None:
-                self._state = round(self.fido_data.data[self.type], 2)
+        if self._name == 'balance':
+            if self.type in self.fido_data.data:
+                if self.fido_data.data[self.type] is not None:
```

Comment:
You can only use `if self.fido_data.data.get(self.type):` that replace this two lines.